cmd/bosun/expr/funcs.go

```diff
@@ -691,11 +692,16 @@ func timeTSDBRequest(e *State, T miniprofiler.Timer, req *opentsdb.Request) (s o
 	return
 }
 
-func timeLSRequest(e *State, T miniprofiler.Timer, service *elastic.SearchService, source *elastic.SearchSource) (resp *elastic.SearchResult, err error) {
-	e.logstashQueries = append(e.logstashQueries, *service.SearchSource(source))
-	b, _ := json.MarshalIndent(source.Source(), "", "  ")
+func timeLSRequest(e *State, T miniprofiler.Timer, req *logstash.Request) (resp *elastic.SearchResult, err error) {
+	e.logstashQueries = append(e.logstashQueries, *req.Source)
+	b, _ := json.MarshalIndent(req.Source.Source(), "", "  ")
 	T.StepCustomTiming("logstash", "query", string(b), func() {
-		resp, err = service.SearchSource(source).Do()
+		getFn := func() (interface{}, error) {
+			return e.logstashContext.Query(req)
+		}
+		var val interface{}
+		val, err = e.cache.Get(string(b), getFn)
```

Comment:
@Dieterbe Do you know, I don't have a great grasp on how the caching works..... ?